libraries/loader/src/loader.js

```diff
@@ -61,142 +54,76 @@ function renderGraph(componentDirectory, sortedComponents) {
 
 /*
  * Construct a component loader function.
- *
- * The `componentDirectory` is an object mapping from component names to
- * component loaders as follows:
- * ```js
- * let load = loader({
- *   // Component loader that always returns 'test'
- *   profile: {
- *     setup: () => 'test'
- *   },
- *
- *   // Component loader that requires profile as input to the setup function
- *   config: {
- *     requires: ['profile'],
- *     setup: (options) => {
- *       return base.config({profile: options.profile});
- *     }
- *   },
- *
- *   // Component loader that loads asynchronously
- *   requestedValue: {
- *     requires: ['config'],
- *     setup: async (options) => {
- *       let res = await request.get(config.some_url).get().end();
- *       return res.body;
- *     }
- *   },
- *
- *   // Component loader that requires more than one component
- *   server: {
- *     requires: ['config', 'requestedValue'],
- *     setup: (options) => {
- *       return server.startListening({
- *         config: options.config,
- *         input: options.requestedValues,
- *       });
- *     }
- *   }
- * });
- * ```
- * With this `load` function you can load the server using:
- * ```js
- * let server = await load('server');
- * ```
- * Naturally, you can also load config `await load('config');` which is useful
- * for testing.
- *
- * Sometimes it's not convenient to hard code constant values into the component
- * directory, in the example above someone might want to load the
- * components with a different profile. Instead you can specify "profile" as
- * a `virtualComponents`, then it must be provided as an options when loading.
- *
- * ```js
- * let load = loader({
- *   // Component loader that requires profile as input to the setup function
- *   config: {
- *     requires: ['profile'],
- *     setup: (options) => {
- *       return base.config({profile: options.profile});
- *     }
- *   }
- * }, ['profile']);
- * ```
- *
- * Then you'll be able to load config as:
- * ```js
- * let config = await load('config', {profile: 'test'});
- * ```
+ * Usage is detailed in README.
  */
-function loader(componentDirectory, virtualComponents = []) {
```

Comment:
Rather than breaking backwards compatibility, what about taking either a list or an object and converting lists into null value objects?

```javascript
> let x = {}
undefined
> ['port'].forEach(y => x[y] = null)
undefined
> x
{ port: null }
>
```

Reply:
Nice. Will update that to catch any cases I missed while updating.